numpy/tests/test_public_api.py

```diff
@@ -746,3 +747,67 @@ def test___module__attribute():
 
     if incorrect_entries:
         assert len(incorrect_entries) == 0, incorrect_entries
+
+
+def _check___qualname__(obj, module_name: str) -> bool:
+    qualname = obj.__qualname__
+    name = obj.__name__
+    assert name == qualname.split(".")[-1]
+
+    # trimcoef is re-exported in a few modules
+    if name == "trimcoef":
+        module_name = obj.__module__
```

Comment:
Ah, I thought to use this unconditionally, because that way we have an explicit test that `obj.__module__:obj.__qualname__` works (I am not sure if we test that right now).